tests/tests_app/utilities/test_proxies.py

```diff
@@ -644,56 +645,71 @@ def test_state_observer():
 )
-def test_work_runner_sets_internal_ip(environment, expected_ip_addr):
+def test_work_runner_sets_internal_ip(patch_constants, environment, expected_ip_addr):
     """Test that the WorkRunner updates the internal ip address as soon as the Work starts running."""
+    # Set constants
+    old_constants = {}
+    for constant, value in patch_constants.items():
+        old_constants[constant] = getattr(constants, constant)
+        setattr(constants, constant, value)
 
-    class Work(LightningWork):
-        def run(self):
-            pass
+    try:
 
-    work = Work()
-    work_runner = WorkRunner(
-        work,
-        work.name,
-        caller_queue=_MockQueue("caller_queue"),
-        delta_queue=Mock(),
-        readiness_queue=Mock(),
-        error_queue=Mock(),
-        request_queue=Mock(),
-        response_queue=Mock(),
-        copy_request_queue=Mock(),
-        copy_response_queue=Mock(),
-    )
+        class Work(LightningWork):
+            def run(self):
+                pass
 
-    # Make a fake call
-    call_hash = "run:fe3fa0f34fc1317e152e5afb023332995392071046f1ea51c34c7c9766e3676c"
-    work._calls[call_hash] = {
-        "args": (),
-        "kwargs": {},
-        "call_hash": call_hash,
-        "run_started_counter": 1,
-        "statuses": [],
-    }
-    work_runner.caller_queue.put(
-        {
+        work = Work()
+        work_runner = WorkRunner(
+            work,
+            work.name,
+            caller_queue=_MockQueue("caller_queue"),
+            delta_queue=Mock(),
+            readiness_queue=Mock(),
+            error_queue=Mock(),
+            request_queue=Mock(),
+            response_queue=Mock(),
+            copy_request_queue=Mock(),
+            copy_response_queue=Mock(),
+        )
+
+        # Make a fake call
+        call_hash = "run:fe3fa0f34fc1317e152e5afb023332995392071046f1ea51c34c7c9766e3676c"
+        work._calls[call_hash] = {
             "args": (),
             "kwargs": {},
             "call_hash": call_hash,
-            "state": work.state,
+            "run_started_counter": 1,
+            "statuses": [],
         }
-    )
-
-    with mock.patch.dict(os.environ, environment, clear=True):
-        work_runner.setup()
-        # The internal ip address only becomes available once the hardware is up / the work is running.
-        assert work.internal_ip == ""
-        try:
-            work_runner.run_once()
-        except Empty:
-            pass
-        assert work.internal_ip == expected_ip_addr
+        work_runner.caller_queue.put(
+            {
+                "args": (),
+                "kwargs": {},
+                "call_hash": call_hash,
+                "state": work.state,
+            }
+        )
+
+        with mock.patch.dict(os.environ, environment, clear=True):
+            work_runner.setup()
+            # The internal ip address only becomes available once the hardware is up / the work is running.
+            assert work.internal_ip == ""
```

Comment:
Again, how this test works when it is under trying...

Reply:
Changed for fixture